File: javascript/codeql/command_line_injection/malicious_edge.js
const express = require("express");
const childProcess = require("child_process");

const app = express();

function buildCommand(host) {
  return "ping -c 1 " + host;
}

app.get("/ping-helper", (req, res) => {
  childProcess.exec(buildCommand(req.get("X-Target-Host") || "127.0.0.1"), (error, stdout) => {
    res.status(error ? 500 : 200).send(stdout);
  });
});

File: python/codeql/command_line_injection/malicious_edge.py
import subprocess

from flask import Flask, request


app = Flask(__name__)


def build_command(host: str) -> str:
    return "ping -c 1 " + host


@app.get("/ping-helper")
def ping_helper():
    command = build_command(request.headers.get("X-Target-Host", "127.0.0.1"))
    return subprocess.check_output(command, shell=True, text=True)
